NextGenConnect/JSON to HL7v2.xml

```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
 }
 tmp[&apos;PID&apos;][&apos;PID.3&apos;] = ids.join(&apos;~&apos;);
 
-
-/*
 //PID-5 patient name
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.1&apos;] = patient[&apos;name&apos;][0][&apos;family&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.3&apos;] = &quot;&quot;; //not in XML
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.1&apos;] = patient[&apos;name&apos;][0][&apos;family&apos;];
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][0]; //Why is this stored in an array in sample JSON message?
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.7&apos;] = &quot;L&quot;;  // Legal Name		
+
+//PID-6 Mothers Maiden Name -- ??? Try to grab value ?
+
+//PID-7 Birthdate 
+
+var birthDate = patient[&apos;birthDate&apos;];
+
+birthDate.replace(&apos;-&apos;,&apos;&apos;); // Replaces -&apos;s in birthDate
```

Comment:
Also, I'm not sure why, but dashes were included in the HL7 message when they shouldn't be. Maybe .replace doesn't work in-place in javascript.

Reply:
It probably doesn't I'm going to look for a different function

Reply:
This is done.